envconfig/src/utils.rs

```diff
@@ -4,19 +4,20 @@ use std::str::FromStr;
 use crate::error::Error;
 use std::collections::HashMap;
 
-/// Load a nenvironment variable by name and parse it into type `T`.
-/// The function is used by `envconfig_derive` to implement `init()`.
+/// Load an environment variable by name and parse it into type `T`.
 ///
-/// It returns `Error` in the following cases:
+/// This function can also use a hashmap as a fallback or for testing purposes.
+///
+/// # Errors
 /// - Environment variable is not present
 /// - Parsing failed
-pub fn load_var<T: FromStr>(
+pub fn load_var<T: FromStr, S: ::std::hash::BuildHasher>(
```

Comment:
I wonder, what's your reasoning to make it generic over hasher? (I guess it's not often when end user would care about this)

Reply:
It was a recommendation by clippy.
Honestly I thought that this doesn't do any harm and if it is good practice according to clippy: why not.